test/app/lib/CoursesHelper.spec.coffee

```diff
@@ -0,0 +1,115 @@
+helper = require 'lib/coursesHelper'
+Campaigns = require 'collections/Campaigns'
+Users = require 'collections/Users'
+Courses = require 'collections/Courses'
+CourseInstances = require 'collections/CourseInstances'
+Classrooms = require 'collections/Classrooms'
+
+describe 'CoursesHelper', ->
+
+  describe 'calculateAllProgress', ->
+
+    beforeEach ->
+      # classrooms, courses, campaigns, courseInstances, students
+      @classrooms = require 'test/app/fixtures/classrooms'
+      @classroom = @classrooms.models[0]
+      @courses = require 'test/app/fixtures/courses'
+      @course = @courses.models[0]
+      @campaigns = require 'test/app/fixtures/campaigns'
+      @campaign = @campaigns.models[0]
+      @students = require 'test/app/fixtures/students'
+
+    describe 'when all students have completed a course', ->
+
+      beforeEach ->
+        @classroom.sessions = require 'test/app/fixtures/level-sessions-completed'
+        @courseInstances = require 'test/app/fixtures/course-instances'
+
+      it 'progressData.get({classroom, course}) returns object with .completed=true and .started=true', ->
```

Comment:
Would you duplicate the `when all students have completed a course` under four separate describes for each of the ways to call `progressData.get`?